API/jsi/jsi/jsi.cpp

```diff
@@ -280,7 +288,17 @@ Instrumentation& Runtime::instrumentation() {
   return sharedInstance;
 }
 
-Value Runtime::createValueFromJsonUtf8(const uint8_t* json, size_t length) {
+// Weak to allow other (external) libraries to provide an implementation.
+JSI_WEAK void createNodeApiEnvFallback(int32_t apiVersion, napi_env *) {
```

Comment:
Providing a weak fallback implementation to allow other (external) libraries to provide an implementation.

Reply:
I wonder if the weak symbols work correctly in MSVC.

Reply:
I pushed changes using the pattern above and the tests are now passing 👍 Let me know if I've missed something or it could be simplified further.

The weak symbols were used not only for `createNodeApiEnvFallback` but also for `createNodeApiEnv` getting its implementation overridden from `hermesNodeApi`. I'd suggest we keep it simple and not make it compiler specific, except if the Hermes team pushes back on this.